x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/get_dataset_info/get_relevant_field_names.ts

```diff
@@ -106,8 +106,8 @@ export async function getRelevantFieldNames({
             types provided in the last user message. DO NOT UNDER ANY
             CIRCUMSTANCES include fields not mentioned in this list.`,
           messages: [
-            // remove the function request
-            ...messages.filter((msg) => !msg.message?.function_call),
+            // remove the last function request
+            ...messages.slice(0, -1),
```

Comment:
This was the root cause. The rest of the PR is just adding logs to aid in debugging